sqlglot/dialects/snowflake.py

```diff
@@ -216,11 +262,13 @@ class Generator(generator.Generator):
             exp.Parameter: lambda self, e: f"${self.sql(e, 'this')}",
             exp.PartitionedByProperty: lambda self, e: f"PARTITION BY {self.sql(e, 'this')}",
             exp.Matches: rename_func("DECODE"),
+            exp.Select: _select_sql,
```

Comment:
move these private methods into the Generator and just override the methods

def select_sql(...)